client/src/main/scala/org/zalando/nakadi/client/Client.scala

```diff
@@ -0,0 +1,218 @@
+package org.zalando.nakadi.client
+
+import scala.concurrent.Future
+import akka.actor.Terminated
+import akka.http.scaladsl.model.HttpResponse
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.Logger
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.protobuf.ByteString
+import org.zalando.nakadi.client.model.EventType
+import akka.http.scaladsl.unmarshalling._
+import org.zalando.nakadi.client.model.DefaultMarshaller
+
+trait Client {
+
+  /**
+   * Retrieve monitoring metrics
+   *
+   * {{{
+   * curl --request GET /metrics
+   * }}}
+   */
+  def metrics(): Future[HttpResponse]
+
+  /**
+   * Returns a list of all registered EventTypes.
+   *
+   * {{{
+   * curl --request GET /event-types
+   * }}}
+   *
+   */
+  def eventTypes()(implicit marshaller: FromEntityUnmarshaller[EventType], charset: String = "UTF-8"): Future[EventType]
+
+  /**
+   * Creates a new EventType.
+   *
+   * {{{
+   * curl --request POST -d @fileWithEventType /event-types
+   * }}}
+   *
+   * @param event - The EventType to create.
+   *
+   */
+  def newEventType(eventType: Any): Future[Boolean]
+
+  /**
+   * Returns the EventType identified by its name.
+   * {{{
+   * curl --request GET /event-types/{name}
+   * }}}
+   * @param name - Name of the EventType
+   */
+  def eventType(name: String): Future[HttpResponse]
+  /**
+   * Updates the EventType identified by its name.
+   * {{{
+   * curl --request PUT -d @fileWithEventType /event-types/{name}
+   * }}}
+   * @param name - Name of the EventType
+   * @param event - Event to update
+   */
+  def updateEventType(name: String, event: Any): Future[HttpResponse]
+  /**
+   * Deletes an EventType identified by its name.
+   *
+   * {{{
+   * curl --request DELETE /event-types/{name}
+   * }}}
+   *
+   * @param name - Name of the EventType
+   */
+  def deleteEventType(name: String): Future[HttpResponse]
+
+  /**
+   * Creates a new Event for the given EventType.
+   * {{{
+   * curl --request POST -d @fileWithEvent /event-types/{name}/events
+   * }}}
+   * @param name - Name of the EventType
+   * @param event - Event to create
+   */
+  def newEvent(name: String, event: Any): Future[HttpResponse]
+
+  /**
+   * Request a stream delivery for the specified partitions of the given EventType.
+   * {{{
+   * curl --request GET  /event-types/{name}/events
+   * }}}
+   * @param name - Name of the EventType
+   *
+   */
+  def events(name: String): Future[HttpResponse]
```

Comment:
I think this won't work for streaming because you have a connection which is kept open from which you continuously receive events. I do not see how the client could receive the events with this methods